src/pal/src/include/pal/stackstring.hpp

```diff
@@ -44,34 +52,42 @@ class StackString
             return;
         }
 
+        if(preserve)
```

Comment:
It looks like we can support it with code that still looks reasonable. How about something like this?:
```C++
    if (preserve && (m_innerBuffer != m_buffer))
    {
        newBuffer = (T *)PAL_realloc(m_innerBuffer, (count_allocated + 1) * sizeof(T));    
    }
    else
    {
        newBuffer = (T *)PAL_malloc((count_allocated + 1) * sizeof(T));
        if (NULL != newBuffer)
        {
            if (reserve)
            {
                CopyMemory(newBuffer, m_buffer, (m_count + 1) * sizeof(T));
            }
            DeleteBuffer();
        }
    }

    if (NULL == newBuffer)
    {
        SetLastError(ERROR_NOT_ENOUGH_MEMORY);
        return;
    }   
```